[WPE][GTK] WEBKIT_PLUGIN_ERROR_WILL_HANDLE_LOAD is gone, but still required

carlosgcampos opened the pull request:
#### cb508f4d82bfa3073355278c7c0a08cc89256468
<pre>
[WPE][GTK] WEBKIT_PLUGIN_ERROR_WILL_HANDLE_LOAD is gone, but still required
<a href="https://bugs.webkit.org/show_bug.cgi?id=250023">https://bugs.webkit.org/show_bug.cgi?id=250023</a>

Reviewed by Michael Catanzaro.

Add new error domain WebKitMediaError and code WEBKIT_MEDIA_ERROR_WILL_HANDLE_LOAD.

* Source/WebKit/Shared/API/APIError.cpp:
(API::Error::webKitPluginErrorDomain):
* Source/WebKit/UIProcess/API/glib/WebKitError.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitError.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitError):
(toWebCoreError):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewLoadFail):

Canonical link: <a href="https://commits.webkit.org/258772@main">https://commits.webkit.org/258772@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0d95a1438dcbe6ef4b7f2f735ba5b72b697b4965

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102770 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11891 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35798 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112027 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12903 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2800 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95023 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109707 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108545 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9901 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93115 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37550 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/91753 "Found 1 new API test failure: TestIPC.StreamConnectionTest.SendAsyncReplyMessage (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24635 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/79289 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5346 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26052 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5496 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2499 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11515 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45544 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7238 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3210 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Good. Only one remaining problem: we surely have to un-deprecate WebKitPluginError now, because we cannot add WebKitMediaError to the old API without breaking existing applications. Right?

carlosgcampos committed:
[WPE][GTK] WEBKIT_PLUGIN_ERROR_WILL_HANDLE_LOAD is gone, but still required
https://bugs.webkit.org/show_bug.cgi?id=250023

Reviewed by Michael Catanzaro.

Add new error domain WebKitMediaError and code WEBKIT_MEDIA_ERROR_WILL_HANDLE_LOAD.

* Source/WebKit/Shared/API/APIError.cpp:
(API::Error::webKitPluginErrorDomain):
* Source/WebKit/UIProcess/API/glib/WebKitError.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitError.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitError):
(toWebCoreError):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewLoadFail):

Canonical link: https://commits.webkit.org/258772@main
